pennylane/transforms/commutation_dag.py

```diff
@@ -0,0 +1,1049 @@
+    ):
+        return qml.RY(u3.data[0], wires=u3.wires)
+
+    return u3
+
+
+def simplify(operation):
+    r"""Simplify a rotation/ controlled rotation into RX, CRX, RY, CRY, RZ, CZ, H and CH.
+
+    Args:
+        operation (pennylane.Operation): Rotation or controlled rotation.
+
+    Returns:
+         qml.operation: Simplified rotation if possible.
+    """
+    if operation.name not in ["Rot", "U2", "U3", "CRot"]:
+        raise qml.QuantumFunctionError(f"{operation.name} is not a Rot, U2, U3 or CRot.")
+
+    if operation.name == "Rot":
+        return simplify_rotation(operation)
+
+    if operation.name == "U2":
+        return simplify_u2(operation)
+
+    if operation.name == "U3":
+        return simplify_u3(operation)
+
+    return simplify_controlled_rotation(operation)
+
+
+def two_non_simplified_crot(operation1, operation2):
+    r"""Check commutation for two CRot that were not simplified.
+
+    Args:
+        operation1 (pennylane.Operation): First operation.
+        operation2 (pennylane.Operation): Second operation.
+
+    Returns:
+         Bool: True if commutation, False otherwise.
+    """
+    # Two non simplified CRot
+    target_wires_1 = qml.wires.Wires(
+        [w for w in operation1.wires if w not in operation1.control_wires]
+    )
+    target_wires_2 = qml.wires.Wires(
+        [w for w in operation2.wires if w not in operation2.control_wires]
+    )
+
+    control_control = intersection(operation1.control_wires, operation2.control_wires)
+    target_target = intersection(target_wires_1, target_wires_2)
+
+    if control_control and target_target:
+        return np.all(
+            np.allclose(
+                np.matmul(operation1.get_matrix(), operation2.get_matrix()),
+                np.matmul(operation2.get_matrix(), operation1.get_matrix()),
+            )
+        )
+    elif control_control and not target_target:
+        return True
+    elif not control_control and target_target:
+        return np.all(
+            np.allclose(
+                np.matmul(
+                    qml.Rot(*operation1.data, wires=operation1.wires[1]).get_matrix(),
+                    qml.Rot(*operation2.data, wires=operation2.wires[1]).get_matrix(),
+                ),
+                np.matmul(
+                    qml.Rot(*operation2.data, wires=operation2.wires[1]).get_matrix(),
+                    qml.Rot(*operation1.data, wires=operation1.wires[1]).get_matrix(),
+                ),
+            )
+        )
+    return False
+
+
+def simplify_to_identity(operation1, operation2):
+    r"""Check that a parametric operation can be simplified to the identity operator,
+    if it is the case then return commutation relation wiith the second operation.
+
+    Args:
+        operation1 (pennylane.Operation): First operation.
+        operation2 (pennylane.Operation): Second operation.
+
+    Returns:
+         Bool: True if commutation, False otherwise.
+    """
+    if operation1.data and operation1.name != "U2":
```

Comment:
How come U2 is excluded?

Reply:
U2 is the only operation not implementing the identity when all the parameters is 0.